src/main/kotlin/ar/edu/unahur/obj2/seguros/Vehiculo.kt

```diff
@@ -1,27 +1,113 @@
 package ar.edu.unahur.obj2.seguros
 
-abstract class Vehiculo(val valor: Int, val antiguedad: Int) {
+import java.lang.Exception
+import java.time.LocalDate
+
+
+abstract class Vehiculo(val valor: Int, var antiguedad: Int) {
+  val seguros = mutableListOf<Seguro>()
+
+  fun agregarSeguro(seguro: Seguro) = this.seguros.add(seguro)
+
   abstract fun puedeContratar(seguro: Seguro): Boolean
 
+  fun puedeContratarSeguroTerceros() = true
+
+  abstract fun puedeContratarSeguroRoboHurto(): Boolean
+
+  abstract fun puedeContratarSeguroTodoRiesgo(): Boolean
+
+  fun costoPara(seguro: Seguro): Double = seguro.costoPara(this)
+
+  abstract fun costoParaSeguroTerceros():Double
+
+  abstract fun costoParaRoboHurto(): Double
+
+  abstract fun costoParaTodoRiesgo(): Double
+
   fun contratar(seguro: Seguro) {
-    TODO("Not yet implemented")
+    if (this.puedeContratar(seguro))
+      this.agregarSeguro(seguro)
+    else
+      throw Exception ("No puede contratar este seguro :(")
   }
+
+  fun costoTotal() = this.seguros.sumBy { it.costoPara(this).toInt() }
 }
 
 class Auto(valor: Int, antiguedad: Int) : Vehiculo(valor, antiguedad) {
   override fun puedeContratar(seguro: Seguro): Boolean {
-    TODO("Not yet implemented")
+    return seguro.puedeContratar(this)
+  }
+  override fun puedeContratarSeguroRoboHurto()= true
+
+  override fun puedeContratarSeguroTodoRiesgo() = true
+
+  override fun costoParaSeguroTerceros() = (this.valor * 1.0) / 100
+
+  override fun costoParaRoboHurto() = (this.valor * this.porcentajeAntiguedad()) / 100
+
+  override fun costoParaTodoRiesgo() = (this.valor * 7.0) / 100
+
+
+
+  fun porcentajeAntiguedad() = when {
+    this.anioActual() - this.antiguedad >= 1995 -> 3.0
+    else  -> 5.0
   }
+
+  fun anioActual() = LocalDate.now().year
+
+
 }
 
 class Camion(valor: Int, antiguedad: Int) : Vehiculo(valor, antiguedad) {
   override fun puedeContratar(seguro: Seguro): Boolean {
-    TODO("Not yet implemented")
+    return seguro.puedeContratar(this)
   }
+
+  override fun puedeContratarSeguroRoboHurto()= this.antiguedad < 10
+
+  override fun puedeContratarSeguroTodoRiesgo() = false
+
+  override fun costoParaSeguroTerceros() = (this.valor * this.porcentajeAntiguedad()) / 100
+
+  fun porcentajeAntiguedad() = when {
+    this.antiguedad < 10 -> 1.5
+    else -> 2.0
+  }
```

Comment:
Muy bueno, aunque cuando solo son dos casos también se puede hacer con un `if`:

```suggestion
  fun porcentajeAntiguedad() = if (this.antiguedad < 10) 1.5 else 2.0
```